### PR TITLE
Ledger support

### PR DESCRIPTION
**Pull request by polarker**

### CODE REVIEW

File: packages/extension/src/ui/features/ledger/utils.ts [hunk 1]
```diff
@@ -24,8 +26,35 @@ export const getLedgerApp = async () => {
...
+  for (let i = 0; i < relevantAccounts.length; i++) {
+    const existingAccount = relevantAccounts[i]
+    const path = getHDWalletPath(keyType, existingAccount.signer.derivationIndex + 1)
+    console.log(`===== path`, path)
+    const [newAccount, hdIndex] = await app.getAccount(path, targetAddressGroup, keyType)
+    if (existingLedgerAccounts.find((account) => account.address === newAccount.address) === undefined) {
+      return [newAccount, hdIndex] as const
+    }
+  }
+  const path = getHDWalletPath(keyType, 0)
```
Comment (h0ngcha0) [inline packages/extension/src/ui/features/ledger/utils.ts:56]: Q: with the for loop above, is it possible that we end up here?

File: packages/extension/src/ui/features/accounts/accountMetadata.state.ts [hunk 2]
```diff
@@ -41,7 +41,8 @@ export const setDefaultAccountNames = (accounts: Account[]) => {
   for (const account of accounts) {
     const { networkId } = account
     if (!names[networkId]?.[account.address]) {
-      const name = `Account ${
+      const prefix = account.signer.type === "ledger" ? "Ledger " : "Account"
```
Comment (h0ngcha0) [inline packages/extension/src/ui/features/accounts/accountMetadata.state.ts:44]: [Nit] is it intentional to have a space after `Ledger` but not `Account`?

**Review by h0ngcha0 — COMMENTED** [review]
Great work, only a few comments and questions

File: packages/extension/src/background/accountMessaging.ts [hunk 3]
```diff
@@ -68,6 +68,36 @@ export const handleAccountMessage: HandleMessage<AccountMessage> = async ({
...
+          data: {
+            account: baseAccount
+          }
+        })
+      } catch (exception) {
+        const error = `${exception}`
+
+        analytics.track("createAccount", {
+          status: "failure",
+          networkId: networkId,
```
[the remark this answers is not in the document]
Comment (polarker) [reply packages/extension/src/background/accountMessaging.ts:90]: Probably the information should be in the error message. Right now, we don't store any other metadata, e.g. hd path, key type etc.

File: packages/extension/src/ui/features/ledger/start.tsx [hunk 4]
```diff
@@ -25,12 +24,28 @@ export const StyledButton = styled(Button)`
 `
 
 export const LedgerStartScreen: FC = () => {
+  const { networkId, addressGroupRaw } = useParams()
+  const [addressGroup, setAddressGroup] = useState<number>()
   const navigate = useNavigate()
   const [error, setError] = useState("")
   const [detecting, setDetecting] = useState(false)
-  const { addAccount } = useAddLedgerAccount()
+  const { addAccount } = useAddLedgerAccount(networkId ?? 'devnet')
```
[the remark this answers is not in the document]
Comment (polarker) [reply packages/extension/src/ui/features/ledger/start.tsx]: This is not required now.

File: packages/extension/src/ui/features/ledger/utils.ts (hunk 1 above)
Comment (polarker) [reply packages/extension/src/ui/features/ledger/utils.ts:56]: Yes, it is possible because the number of relevant accounts might be 0

**Pull request merged**